src/ruler.js

```diff
@@ -37,6 +49,7 @@ export class DragRulerRuler extends Ruler {
 			const isToken = this.draggedEntity instanceof Token;
 			json["draggedEntityIsToken"] = isToken;
 			json["draggedEntity"] = this.draggedEntity.id;
+			json["snappedToGrid"] = this.snappedToGrid;
```

Comment:
Is there any reason why other clients would need to know about this? After all, the waypoints that get transmitted are already in their respective snapped or unsnapped position, so what does the other side do with that information? Am I missing something?

Reply:
This change affects how the destination is snapped on remote clients.

**Vanilla Drag Ruler:** This change allows the destination of the ruler to appear unsnapped on remote clients when the local client is holding shift to unsnap. Previously remote clients would see the destination of the ruler snapped to the grid even though that is not what the local client is seeing. This was a nice bonus side effect of the Toggle Snap to Grid integration :)
With my change:
![RemotePlayerUnsnap](https://user-images.githubusercontent.com/25129246/127089426-6b179061-8f5e-4713-8cd0-b229b2a58fe8.png)

**With Toggle Snap to Grid installed:** Because of the "hold shift to re-snap" feature my module needs to know if `snap` is currently `false` from holding down shift so it can correctly re-activate it before calling `measure()`. I got this working on local clients but saw that remote clients were not re-snapping because they didn't know if the local client was holding shift or not. This change tells remote clients if the local client is currently snapping to grid and allows my module to apply the correct state.

Reply:
I just went back and tested this and non-snapping behavior is working without this change - with and without Toggle Snap to Grid enabled. It only seems to break when Toggle Snap to Grid is enabled and the snapping option is enabled for the token. Since all other constellations work, I suppose that that behavior is actually a bug that occurs somewhere in the existing codepath that could be fixed without actually transmitting this additional info.

Reply:
Just tested it as well and it was partly a bug on my end. My Ruler.measure wrapper was always pushing a default snap state equal to the current grid type. I fixed that, though for compatibility we might still need this change.

Without this change, Drag Ruler sends nothing to remote users about snapping, so options.snap is always undefined when we get to Ruler.measure. That works in our favor because each user is sent the exact coordinate the ruler is pointing to, which the sending client will have already snapped to the grid or left unsnapped if shift is held. Remote clients assume snapping is off at all times and just use whatever coordinate is sent.

Complications appear when my module's logic takes an inputted snap setting and uses that to determine what output to send off to Ruler.measure. A couple of example cases, keep in mind remote clients always assume snapping is disabled:
- Local client has snapping toggled off and is dragging the token normally. My module on the remote client sees `options.snap != true`, assumes shift is being held by the local client, and re-enables snapping. Remote clients see the token snapped to the grid (wrong behavior).
- Local client has snapping toggled off and is holding shift while dragging the token. My module acts the same as above (correct behavior).

Basically, I need some way to know if remote clients are snapping to the grid. Best I can think of is sending that boolean over the network. You have another idea?

Reply:
Actually just had some ideas. Gonna sleep on it and try some stuff out

Reply:
In my testing it appears that the state of the "toggle snap to grid" box is already synchronized between the clients. Couldn't you use that value to determine the snapping behavior?

Reply:
Pushed an update to my module that removes the need for these changes. Just removed them from this PR as well.

Now it works the same as if my module was uninstalled: a remote client's ruler always has options.snap != true. The destination and all of the waypoints are networked over in the correct state so there isn't a need to know how to snap them. Had to remove that option that allowed reverting to the map's measurement/highlighting when holding shift, but I wasn't ever too attached to it even if that's how I preferred it. Holding shift now just resnaps the token and doesn't modify any more than that.